controllers/v1/solutions.js

```diff
@@ -16,64 +16,1066 @@ module.exports = class  Solutions extends Abstract{
    * @returns {JSON} Created solution data.
    */
 
-  async create(req) {
+  async detailsBasedOnRoleAndLocation(req) {
     return new Promise(async (resolve, reject) => {
       try {
 
-        let createdSolution = await solutionsHelper.create(
+        let solutionDetails = 
+        await solutionsHelper.detailsBasedOnRoleAndLocation(
+          req.params._id,
           req.body,
-          req.userDetails.userToken
+          req.query.type ? req.query.type : ""
+        );
+          
+        return resolve(solutionDetails);
+
+      } catch (error) {
+        return reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        });
+      }
+    });
+  }
+
+       /**
+    * @api {post} /project/v1/solutions/removeRolesInScope/:solutionId Remove roles from solutions scope
+    * @apiVersion 1.0.0
+    * @apiName 
+    * @apiGroup Solutions
+    * @apiParamExample {json} Request-Body:
+    * {
+    * "roles" : ["DEO","SPD"]
+    }
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
+    * @apiSampleRequest /project/v1/solutions/removeRolesInScope/5ffbf8909259097d48017bbf
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+        "message": "Successfully removed roles in solution scope",
+        "status": 200
+      }
+    */
+
+     /**
+   * Remove roles in solution scope
+   * @method
+   * @name removeRolesInScope
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - solution id.
+   * @param {Array} req.body.roles - Roles to be added.
+   * @returns {Array} Removed solution scope roles.
+   */
+
+     async removeRolesInScope(req) {
+      return new Promise(async (resolve, reject) => {
+        try {
+  
+          let solutionUpdated = await solutionsHelper.removeRolesInScope(
+            req.params._id,
+            req.body.roles
+          );
+      
+          return resolve(solutionUpdated);
+  
+        } catch (error) {
+          return reject({
+            status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+            message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+            errorObject: error
+          });
+        }
+      });
+    }
+
+
+
+   /**
+  * @api {get} /project/v1/solutions/fetchLink/:solutionId
+  * @apiVersion 1.0.0
+  * @apiName Get link by solution id
+  * @apiGroup Solutions
+  * @apiSampleRequest /project/v1/solutions/fetchLink/5fa28620b6bd9b757dc4e932
+  * @apiHeader {String} X-authenticated-user-token Authenticity token  
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  * {
+    "message": "Solution Link generated successfully",
+    "status": 200,
+    "result": "https://dev.sunbirded.org/manage-learn/create-observation/38cd93bdb87489c3890fe0ab00e7d406"
```

Comment:
@Prajwal17Tunerlabs change create-observation to create-project

Reply:
changed create-observation to create-project